clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -1128,16 +1134,33 @@ resetPolarityProxy _proxy =
 -- loops. In case of synchronous resets it can lead to
 -- <Clash-Explicit-Signal.html#metastability meta-stability> in the presence of
 -- asynchronous resets.
-unsafeToHighPolarity
+unsafeToActiveHigh
   :: forall dom
    . KnownDomain dom
   => Reset dom
   -> Signal dom Bool
-unsafeToHighPolarity (unsafeFromReset -> r) =
+unsafeToActiveHigh (unsafeFromReset -> r) =
   case resetPolarityProxy (Proxy @dom) of
     SActiveHigh -> r
     SActiveLow -> not <$> r
-{-# INLINE unsafeToHighPolarity #-}
+{-# INLINE unsafeToActiveHigh #-}
+
+-- | Convert a reset to an active high reset. Has no effect if reset is already
+-- an active high reset. Is unsafe because it can introduce:
+--
+-- * <Clash-Explicit-Signal.html#metastability meta-stability>
+--
+-- For asynchronous resets it is unsafe because it can cause combinatorial
+-- loops. In case of synchronous resets it can lead to
+-- <Clash-Explicit-Signal.html#metastability meta-stability> in the presence of
+-- asynchronous resets.
+unsafeToHighPolarity
+  :: forall dom
+   . KnownDomain dom
+  => Reset dom
+  -> Signal dom Bool
+unsafeToHighPolarity = unsafeToActiveHigh
+{-# DEPRECATED unsafeToHighPolarity "Use 'unsafeToActiveHigh' instead. This function will be removed in Clash 1.12." #-}
```

Comment:
I think `INLINE unsafeToHighPolarity` is redundant, but if you do add it you don't need to think, it'll just inline for sure.